### PR TITLE
hw/bus: Multiple fixes for bus driver (and related code)

### PR DESCRIPTION
**Pull request by andrzej-kaczmarek**

### CODE REVIEW

File: hw/bus/include/bus/bus.h [hunk 1]
```diff
@@ -210,6 +218,17 @@ bus_node_lock(struct os_dev *node, os_time_t timeout);
 int
 bus_node_unlock(struct os_dev *node);
 
+/**
+ * Get node configured lock timeout
+ *
+ * Returns lock timeout as configured for node. If not timeout is configured for
```
Comment (mkiiskila) [inline hw/bus/include/bus/bus.h]: Spelling: If not timeout is... -> If no timeout is...

File: hw/bus/src/bus.c [hunk 2]
```diff
@@ -200,7 +202,7 @@ bus_node_write_read_transact(struct os_dev *node, const void *wbuf,
         return SYS_ENOTSUP;
     }
 
-    rc = bus_node_lock(node, timeout);
+    rc = bus_node_lock(node, g_bus_node_lock_timeout);
```
Comment (mkiiskila) [inline hw/bus/src/bus.c]: bus_node_get_lock_timeout()?

File: hw/bus/src/bus.c [hunk 3]
```diff
@@ -172,7 +174,7 @@ bus_node_write(struct os_dev *node, const void *buf, uint16_t length,
...
```
Comment (mkiiskila) [inline hw/bus/src/bus.c]: bus_node_get_lock_timeout()?

File: hw/bus/src/bus.c [hunk 4]
```diff
@@ -145,7 +147,7 @@ bus_node_read(struct os_dev *node, void *buf, uint16_t length,
...
```
Comment (mkiiskila) [inline hw/bus/src/bus.c]: bus_node_get_lock_timeout()?

File: hw/bus/src/bus.c [hunk 5]
```diff
@@ -285,8 +292,18 @@ bus_node_unlock(struct os_dev *node)
     return 0;
 }
 
+os_time_t
+bus_node_get_lock_timeout(struct os_dev *node)
```
Comment (mkiiskila) [inline hw/bus/src/bus.c]: I assume this is not the final implementation, and that bus_node_set_lock_timeout() is going to follow?

Comment (andrzej-kaczmarek) [reply hw/bus/src/bus.c]: my idea here is to let user specify per-node lock timeout in node configuration and getter is to return effective lock timeout for a node, i.e. either default one (if not configured) or the one configured for a node. I think there's no reason for user to reconfigure lock timeout in runtime - by default all nodes can be locked with the same /reasonable/ timeout but perhaps some nodes should be accessed with e.g. very short lock timeout so they either can be read instantly or not at all and for this it's enough to configure this on build time. but there may be some actual scenarios which require different behaviour so comments are always welcome :)

I'll include implementation of this in next series of commits I'm working on currently.

**Comment by andrzej-kaczmarek** [comment]
Thanks Marko, I'll merge this now and include fixes for those comments in my next PR with bunch of another fixes/improvements for bus driver (just so I don't modify already approved PR).

**Pull request merged**